packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -62,16 +65,31 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
           }
 
           const url = rawUrl.slice(1, -1)
-          const file = path.resolve(path.dirname(id), url)
-          // Get final asset URL. Catch error if the file does not exist,
-          // in which we can resort to the initial URL and let it resolve in runtime
-          const builtUrl = await fileToUrl(file, config, this).catch(() => {
+          assetResolver ??= config.createResolver({
+            root: config.publicDir,
+            extensions: [],
+            mainFields: [],
+            tryIndex: false,
+            preferRelative: true
+          })
+          const resolved = await assetResolver(url, id)
+          if (!resolved) {
```

Comment:
I think we can optimize this like above, when the url starts with `.`, we can result to the simpler `const file = path.resolve(path.dirname(id), url)` like before.